### DISCUSSION TITLE
No cooldown if no permission

### DISCUSSION THREAD

**Discussion by M0diis** [EssentialsX/Essentials#4704] [Ideas and Feature Suggestions] [answered]
Suggestion is simple - don't put the command on cooldown if the player has no permission to use it as it is.

As for example - my server has keys that can be redeemed via a command for their specific ranks, but some players forget which rank they are and if they use the redeem command for the rank they don't have - it puts that redeem command on cooldown.

**Comment by mdcfe** [comment, marked as answer]
There isn't a reliable way for EssentialsX to know whether someone has permission to run commands from other plugins. The best we could do is check the command map permission (the same one Bukkit checks, popping up a standard Bukkit message if not granted) - but this won't catch the vast majority of use cases, including yours.

I'm considering a new system that would let plugins integrate into the EssentialsX cooldowns more deeply, but this is currently a way off.

For now, you would be better off either creating separate cooldowns for each redeemable key using regular expressions - you can see examples over on the [wiki](https://essentialsx.net/wiki/Command-Cooldowns.html).

**Comment by M0diis** [comment]
Well, it would be easier if there was an option to supply the permission in the config.

```
command-cooldowns:
  feed: 10 
  redeem: 
    cooldown: 10
    permission: some.permission.node
  ```
